test/uniswapTrades.test.ts

```diff
@@ -693,4 +701,350 @@ describe('Uniswap', () => {
+      const uniswapTrade = new UniswapTrade(routerTrade, opts)
+
+      // ensure that we can generate the encoded call parameters
+      const methodParameters = SwapRouter.swapCallParameters(uniswapTrade)
+      expect(hexToDecimalString(methodParameters.value)).to.eq(inputAmount)
+
+      // ensure that the encoded call parameters are the same
+      expect(methodParameters.calldata).to.eq(SDKMethodParameters.calldata)
+    })
+
+    it('v2 - handles eth output properly', async () => {
+      const inputAmount = utils.parseUnits('1000', 6).toString()
+      const rawInputAmount = CurrencyAmount.fromRawAmount(USDC, inputAmount)
+      const opts = swapOptions({})
+      const trade = new V2Trade(
+        new RouteV2([WETH_USDC_V2], USDC, Ether.onChain(1)),
+        rawInputAmount,
+        TradeType.EXACT_INPUT
+      )
+      const SDKMethodParameters = SwapRouter.swapCallParameters(new UniswapTrade(buildTrade([trade]), opts))
+
+      const rawOutputAmount = trade.outputAmount
+      const outputAmount = utils.parseEther(rawOutputAmount.toExact()).toString()
+
+      const classicQuote: PartialClassicQuote = {
+        tokenIn: USDC.address,
+        tokenOut: ETH_ADDRESS,
+        tradeType: TradeType.EXACT_INPUT,
+        route: [
+          [
+            // WETH here since all pairs use WETH
+            mockV2PoolInRoute(WETH_USDC_V2, USDC, WETH, inputAmount, outputAmount),
+          ],
+        ],
+      }
+      const routerTrade = RouterTradeAdapter.fromClassicQuote(classicQuote)
+      const uniswapTrade = new UniswapTrade(routerTrade, opts)
+
+      // ensure that we can generate the encoded call parameters
+      const methodParameters = SwapRouter.swapCallParameters(uniswapTrade)
+      expect(hexToDecimalString(methodParameters.value)).to.eq('0')
+
+      // ensure that the encoded call parameters are the same
+      expect(methodParameters.calldata).to.eq(SDKMethodParameters.calldata)
+    })
+
+    it('v3 - handles eth output properly', async () => {
+      const inputAmount = utils.parseUnits('1000', 6).toString()
+      const rawInputAmount = CurrencyAmount.fromRawAmount(USDC, inputAmount)
+      const opts = swapOptions({})
+      const trade = await V3Trade.fromRoute(
+        new RouteV3([WETH_USDC_V3], USDC, Ether.onChain(1)),
+        rawInputAmount,
+        TradeType.EXACT_INPUT
+      )
+      const SDKMethodParameters = SwapRouter.swapCallParameters(new UniswapTrade(buildTrade([trade]), opts))
+
+      const rawOutputAmount = trade.outputAmount
+      const outputAmount = utils.parseEther(rawOutputAmount.toExact()).toString()
+
+      const classicQuote: PartialClassicQuote = {
+        tokenIn: USDC.address,
+        tokenOut: ETH_ADDRESS,
+        tradeType: TradeType.EXACT_INPUT,
+        route: [
+          [
+            // WETH here since all pairs use WETH
+            mockV3PoolInRoute(WETH_USDC_V3, USDC, WETH, inputAmount, outputAmount),
+          ],
```

Comment:
think some tests for multi hop routes, and mixed routes would be good

Reply:
Just added: exact out, multi hop, split routes and mixed routes